General clean-up

slashrsm opened the pull request.

webflo commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * @return array
+   *   List of css files for inclusion keyed by theme name.
+   */
+  public function css();
```
I guess we need to replace this with libraries? Because its not possible to add individual css files during rendering.

slashrsm replied:
True. Totally slipped out of my mind :)

css() allows to define .css files per theme. Not sure if it makes sense to create entire libraries that are meant to be used on a per-theme basis. Seems kinda wrong. 

I assume the original motivation for this was to be able to ship with CSS for most common admin themes. AFAIK there is still no elegant way to achieve that in D8. Am I correct @bojanz?

Fixed the other comments.

bojanz replied:
It's for per entity type specific styling. We had commerce_product specific styling as the initial use case.

webflo commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * @return
+   *   The entity type.
+   */
+  public function entityType();
```
EntityType or EntityTypeId? I am fine with both but we have to clarify. Maybe similar to EntityInterface?

webflo commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * @param $form_state
+   *   The form state of the parent form.
+   */
+  public function entityFormValidate($entity_form, &$form_state);
```
FormStateInterface?

webflo commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * @param $form_state
+   *   The form state of the parent form.
+   */
+  public function removeForm($remove_form, &$form_state);
```
FormStateInterface?

slashrsm commented:
For now I renamed css() function to libraries() and updated everything to operate with libraries, but still using same per-theme approach. Also reorganized css and js files a bit.

While doing that I noticed that  inline_entity_form.css and inline_entity_form.seven.css are not used anywhere. Safe to remove? Needs update?

Will stop pushing to this branch as I don't want this PR to get to big. Will continue working on separate branches.

marcingy commented on src/Plugin/InlineEntityForm/CommerceProductInlineEntityFormController.php:
```diff
@@ -1,29 +1,33 @@
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
```
Likely on your radar but the factory class for this plugin could be cleaned up by checking for this interface and also by declaring an annotation for the type.

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * Modules can alter the output of this method through
+   * hook_inline_entity_form_table_fields_alter().
+   *
+   * @param $bundles
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+  /**
+   * Returns a setting value.
+   *
+   * @param $name
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * leading to possible key collisions if the keys are not prefixed with
+   * $entity_form['#parents'].
+   *
+   * @param $entity_form
```
Missing type (same with param below)

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+  /**
+   * Validates the entity form.
+   *
+   * @param $entity_form
```
Missing type (same with param below)

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * the values from the form to matching properties and, if the entity is
+   * fieldable, invoking Field API submit.
+   *
+   * @param $entity_form
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+  /**
+   * Returns the remove form to be shown through the IEF widget.
+   *
+   * @param $remove_form
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+   * Handles the submission of a remove form.
+   * Decides what should happen to the entity after the removal confirmation.
+   *
+   * @param $remove_form
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+  /**
+   * Permanently saves the given entity.
+   *
+   * @param $entity
```
Missing type

marcingy commented on src/InlineEntityFormControllerInterface.php:
```diff
@@ -0,0 +1,179 @@
+  /**
+   * Delete permanently saved entities.
+   *
+   * @param $ids
```
Missing type

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -24,94 +27,85 @@
+    }
+    else {
+      $labels = [
+        'singular' => t('entity'),
```
Capital E???

slashrsm replied:
I believe not. This stuff is used in the various sentences, etc.

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -264,18 +254,9 @@ public function entityFormValidate($entity_form, &$form_state) {
+   * {@inheritdoc}
    */
-  public function entityFormSubmit(&$entity_form, FormStateInterface $form_state) {
+  public function entityFormSubmit(&$entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway (same above I see)

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -344,7 +325,7 @@ public function entityFormSubmit(&$entity_form, FormStateInterface $form_state)
    *   The form state of the parent form.
    */
-  protected function cleanupFieldFormState($entity_form, &$form_state) {
+  protected function cleanupFieldFormState($entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -365,14 +346,9 @@ protected function cleanupFieldFormState($entity_form, &$form_state) {
+   * {@inheritdoc}
    */
-  public function removeForm($remove_form, &$form_state) {
+  public function removeForm($remove_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -395,18 +371,9 @@ public function removeForm($remove_form, &$form_state) {
+   * {@inheritdoc}
    */
-  public function removeFormSubmit($remove_form, FormStateInterface $form_state) {
+  public function removeFormSubmit($remove_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -248,14 +243,9 @@ public function entityForm($entity_form, FormStateInterface $form_state) {
+   * {@inheritdoc}
    */
-  public function entityFormValidate($entity_form, &$form_state) {
+  public function entityFormValidate($entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -145,65 +139,66 @@ public function tableFields($bundles) {
+  /**
+   * {@inheritdoc}
+   */
+  public function entityForm($entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -424,43 +391,29 @@ public function removeFormSubmit($remove_form, FormStateInterface $form_state) {
    * @return array
    */
-  protected function buildChildFormState(&$entity_form, &$form_state, $entity, $operation) {
+  protected function buildChildFormState(&$entity_form, FormStateInterface &$form_state, EntityInterface $entity, $operation) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php:
```diff
@@ -424,43 +391,29 @@ public function removeFormSubmit($remove_form, FormStateInterface $form_state) {
   }
 
   /**
    * @param $entity_form
```
Improve doxygen?

marcingy commented on src/Plugin/InlineEntityForm/TaxonomyTermInlineEntityFormController.php:
```diff
@@ -49,9 +54,9 @@ public function tableFields($bundles) {
+   * {@inheritdoc}
    */
-  public function entityForm($entity_form, &$form_state) {
+  public function entityForm($entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

marcingy commented on src/Plugin/InlineEntityForm/TaxonomyTermInlineEntityFormController.php:
```diff
@@ -91,11 +96,10 @@ public function entityForm($entity_form, &$form_state) {
+   * {@inheritdoc}
    */
-  public function entityFormSubmit(&$entity_form, &$form_state) {
+  public function entityFormSubmit(&$entity_form, FormStateInterface &$form_state) {
```
Isn't form_state an object and hence by ref anyway

bojanz commented:
Why are we making inline entity forms into plugins?
For me it seems more logical to rename the controller to handler, and take it from the entity info like in D7.

slashrsm commented:
If we go with entity info we need to hook_info_alter() all core (and other?) entities. Isn't it cleaner this way?

slashrsm commented:
Another pull request based of this branch: https://github.com/slashrsm/inline_entity_form/pull/1

slashrsm commented on src/Plugin/Field/FieldWidget/InlineEntityFormMultiple.php:
```diff
@@ -557,18 +557,14 @@ public function extractFormValues(FieldItemListInterface $items, array $form, Fo
-          // unset($item['entity']->uuid);
-          // }
-          $this->iefController->save($item['entity'], array());
+          $entity->save();
```
I was thinking about removing InlineEntityFormHandlerInterface::save() even before, but decided to postpone it. The thing is that EntityFormInterface (which we need to implement in order to go the form handler way) already defines save() and it's signature conflicts with ours. 

I am aware of the fact that commerce product implementation did some stuff here but I suspect that could also be done via one of the standard core ways (hooks, etc)?

slashrsm commented:
Initial conversion to form handlers done. I confirmed that multiple form widget still works. This needs some more work, such as:
- we should probably convert entityForm() entityFormValidate() and entityFormSubmit() on InlineEntityFormHandlerInterface to FormInterface::buildform() validateForm() and submitForm(), since we indirectly implement the latter interface anyway.
- figure out if we can make initiation of our handler easier (w/o requiring separate configuration step).
- ...

We can do those as follow-ups I guess.

webflo commented on inline_entity_form.module:
```diff
@@ -58,30 +62,18 @@ function inline_entity_form_get_controller(FieldDefinitionInterface $instance) {
-  $entity_info['node']->setFormClass('inline entity form', '\Drupal\inline_entity_form\Plugin\InlineEntityForm\NodeInlineEntityFormController');
+function inline_entity_form_entity_type_build(array &$entity_types) {
+  /** @var \Drupal\Core\Entity\EntityTypeInterface[] $entity_types */
+  $entity_types['node']->setFormClass('inline entity form', '\Drupal\inline_entity_form\InlineEntityForm\NodeInlineEntityFormHandler');
```
I think Node module is optional. We should wrap this and all other core entity types in isset(). Similar to the previous code.

slashrsm replied:
Done.

slashrsm commented:
I guess we can merge this to allow creation of subsequent PRs.
